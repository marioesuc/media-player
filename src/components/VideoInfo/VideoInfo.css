.VideoInfo-container {
  width: 700px;
  height: auto;
  padding: 15px;
  background-color: white;
}

.VideoInfo-description {
  flex: 1;
  height: 85%;
  color: black;
  overflow: scroll;
}
