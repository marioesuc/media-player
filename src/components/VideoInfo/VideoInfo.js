import React from "react";
import "./VideoInfo.css";
import Button from "../Button/Button";

class VideoInfo extends React.Component {
  handlePlayVideo = () => {
    this.props.videoRef.current.play();
    this.props.videoRef.current.requestFullscreen();
  };

  render() {
    return (
      <div className="VideoInfo-container">
        <div className="VideoInfo-description">
          <h2>Description</h2>
          <p>Description</p>
          <h2>Metadata</h2>
          <ul>
            <li>Language</li>
            <li>Release date</li>
          </ul>
        </div>
        <Button onClick={this.handlePlayVideo} />
      </div>
    );
  }
}

export default VideoInfo;
