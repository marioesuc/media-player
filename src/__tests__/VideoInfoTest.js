import React from "react";
import { shallow, configure } from "enzyme";
import VideoInfo from "../components/VideoInfo/VideoInfo";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("VideoInfo", () => {
  const videoRef = React.createRef();

  const videoData = {
    adult: false,
    backdrop_path: "/aQXTw3wIWuFMy0beXRiZ1xVKtcf.jpg",
    genre_ids: (4)[(28, 878, 53, 12)],
    id: 399579,
    original_language: "en",
    original_title: "Alita: Battle Angel",
    overview:
      "When Alita awakens with no memory of who she is in a future world she does not recognize, she is taken in by Ido, a compassionate doctor who realizes that somewhere in this abandoned cyborg shell is the heart and soul of a young woman with an extraordinary past.",
    popularity: 206.119,
    poster_path: "/xRWht48C2V8XNfzvPehyClOvDni.jpg",
    release_date: "2019-01-31",
    title: "Alita: Battle Angel",
    video: false,
    vote_average: 6.7,
    vote_count: 2318
  };

  it("renders correctly", () => {
    const wrapper = shallow(
      <VideoInfo videoRef={videoRef} videoData={videoData} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
