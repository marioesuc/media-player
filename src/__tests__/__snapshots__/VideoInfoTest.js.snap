// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfo renders correctly 1`] = `ShallowWrapper {}`;
